### PR TITLE
Add turbolinks support to has many js

### LINKED ISSUE #7383: hasMany.js does not support turbolinks
### Did you find a bug?

While most other javascript features of ActiveAdmin are set up to be initialized on `page:load` and `turbolinks:load`, the hasMany feature is initialized at run time.

### Expected behavior

Has many forms should still work after turbolinks transitions.

### Actual behavior

The has many initializer is run on page load but not after turbolinks transitions.

### How to reproduce

1. Set up a form with `has_many`
2. Set up turbolinks
3. In the ActiveAdmin UI, create an instance of the resource with has many
4. Load the index page and transition to the edit page with turbolinks enabled
5. Notice that the form on the edit page does not have has many features

### Extra Notes

Hi, I'm working on a PR to fix this now. If anyone wants more detail on this issue, I'm happy to provide it, but I think that the fix will be strait-forward and I just wanted to provide enough context here to support the PR.

Also, it would be nice to support turbo transitions, but that's a job for another PR.

### PR DESCRIPTION
**Pull request by amiel**
This closes #7383.

## Notes

* No regression tests

  There are very little javascript tests, and none around the has many feature. I'm not really sure how to get started but would be happy to look in to it as a followup.

* ~~Formatting changes~~

  ~~My editor automatically made some formatting changes based on the eslint configuration of the project. Would you like me to split these out or remove them?~~

* Turbo vs turbolinks

  I would also like to add support for turbo, and not just turbolinks, but that is also a different project.

### CODE REVIEW

**Comment by deivid-rodriguez** [comment]
Yes, please, no formatting changes! A way to get this tested would be awesome, I'll investigate too. Other than that looks good!

**Comment by amiel** [comment]
@deivid-rodriguez I've updated this to remove the formatting changes. I'll look in to tests as another PR if that's ok with you.

**Comment by deivid-rodriguez** [comment]
I prefer to add tests together with this feature, or at least try :)

File: features/has_many.feature [hunk 1]
```diff
@@ -26,6 +26,7 @@ Feature: Has Many
     When I go to the last author's show page
     And I follow "Edit User"
     Then I should see a link to "Add New Post"
+    And there should be 1 "input" tag within ".ui-sortable"
```
Comment (amiel) [inline features/has_many.feature:29]: I know this is a little awkward, but unfortunately I couldn't find any other observable change for when has many is initialized. The sortable option adds some classes that allowed me to make an assertion that fails before the changes in this PR and passes with the changes.

**Comment by deivid-rodriguez** [comment]
@amiel This looks good to me! Only thing I would change is to completely split the feature test to a separate scenario, so that we can test the `has_many` feature with sortable attribute separately. What do you think?

File: features/has_many.feature (hunk 1 above)
Comment (deivid-rodriguez) [reply features/has_many.feature:29]: It looks good enough to me! Easier that simulating a real drag and drop action.

Comment (amiel) [reply features/has_many.feature:29]: That makes sense. Will do

Comment (amiel) [reply features/has_many.feature]: Ok, how do you feel about this?

**Review by deivid-rodriguez — APPROVED** [review]
It looks perfect to me now, thanks! 🎉 

Only modification I would make is to squash both commits, but that can be done when merging!

**Comment by amiel** [comment]
Squashed and rebased to be current with `master`.

**Comment by deivid-rodriguez** [comment]
Awesome, I'll let @javierjulio double check this if he is available!

Are you planning to also add turbo support, by the way?

**Review by javierjulio — APPROVED** [review]

**Pull request merged**